infrastructure/README.md

```diff
@@ -0,0 +1,7 @@
+This directory contains a number of tests to ensure test running
+infrastructure is operating correctly:
+
+ * The tests in assumptions/ are designed to test UA assumptions
+   documented in ../docs/_writing-tests/assumptions.md.
+
+ * The tests in XFAIL/ should all fail.
```

Comment:
Can you document why there's an X prefix?

Reply:
Is that sufficiently uncommon to justify it? Almost every test framework I've ever touched has a notion of "xfail"?

Reply:
I can't recall every seeing it before. I assumed it was because the string "FAIL" would have a special meaning somewhere and fail a lint, or something.

Reply:
"eXpected fail", I suspect.

Reply:
That's fine, just needs a comment :)

Reply:
Or name it "expected-to-fail" or some such long name. Either is fine if it's documented.

Reply:
fixed